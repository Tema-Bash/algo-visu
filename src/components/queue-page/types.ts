export interface IQueue<T> {
  enqueue: (item: T) => void;
  dequeue: () => void;
  getLength: () => number;
}
