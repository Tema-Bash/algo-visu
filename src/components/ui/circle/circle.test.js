import { fireEvent, render } from '@testing-library/react'
import Circle from './circle.tsx'
import { ElementStates } from '../../../types/element-states'

describe('circle tests', () => {
  it('without letter', () => {
    const container = render(<Circle />)
    const circleLetter = container.getByTestId('letter')
    expect(circleLetter.textContent).toEqual('')
  })

  it('with letter', () => {
    const expectedLetter = 'a'
    const container = render(<Circle letter={expectedLetter} />)
    const circleLetter = container.getByTestId('letter')
    expect(circleLetter.textContent).toEqual(expectedLetter)
    expect(container).toMatchSnapshot()
  })

  it('with head', () => {
    const expectedLetter = 'a'
    const container = render(<Circle head={expectedLetter} />)
    const circleHead = container.getByTestId('head')
    expect(circleHead.textContent).toEqual(expectedLetter)
    expect(container).toMatchSnapshot()
  })

  it('with react element in head', () => {
    const testLetter = 'a'
    const container = render(<Circle head={<Circle letter={testLetter} />} />)
    const circleHead = container.getByText(testLetter)
    expect(circleHead.textContent).toEqual(testLetter)
    expect(container).toMatchSnapshot()
  })

  it('with tail', () => {
    const expectedLetter = 'b'
    const container = render(<Circle tail={expectedLetter} />)
    const circleTail = container.getByTestId('tail')
    expect(circleTail.textContent).toEqual(expectedLetter)
    expect(container).toMatchSnapshot()
  })

  it('with react element in tail', () => {
    const testLetter = 'b'
    const container = render(<Circle tail={<Circle letter={testLetter} />} />)
    const circleTail = container.getByText(testLetter)
    expect(circleTail.textContent).toEqual(testLetter)
    expect(container).toMatchSnapshot()
  })

  it('with index', () => {
    const expectedIndex = '5'
    const container = render(<Circle index={expectedIndex} />)
    const circleIndex = container.getByTestId('index')
    expect(circleIndex.textContent).toEqual(expectedIndex)
    expect(container).toMatchSnapshot()
  })

  it('with prop isSmall', () => {
    const container = render(<Circle isSmall={true} />)
    const smallDiv = container.getByTestId('middleDiv')
    expect(smallDiv).toHaveClass('small')
  })

  it('with state equal default', () => {
    const container = render(<Circle state={ElementStates.Default} />)
    const middleDiv = container.getByTestId('middleDiv')
    expect(middleDiv).toHaveClass('default')
  })

  it('with state equal Changing', () => {
    const container = render(<Circle state={ElementStates.Changing} />)
    const middleDiv = container.getByTestId('middleDiv')
    expect(middleDiv).toHaveClass('changing')
  })

  it('with state equal Modified', () => {
    const container = render(<Circle state={ElementStates.Modified} />)
    const middleDiv = container.getByTestId('middleDiv')
    expect(middleDiv).toHaveClass('modified')
  })
})
