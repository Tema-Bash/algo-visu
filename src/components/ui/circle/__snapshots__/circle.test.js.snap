// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`circle tests with head 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="content "
        data-testid="circle"
      >
        <div
          class="text text_type_input text_color_input mb-4 absolute head string"
          data-cy="head"
          data-testid="head"
        >
          a
        </div>
        <div
          class="circle   default"
          data-cy="circle"
          data-testid="middleDiv"
        >
          <p
            class="text text_type_circle text_color_input letter"
            data-testid="letter"
          />
        </div>
        <p
          class="text text_type_input text_color_input mt-4 absolute index"
          data-testid="index"
        />
        <div
          class="text text_type_input text_color_input mt-4 absolute tail30 element"
          data-cy="tail"
          data-testid="tail"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="content "
      data-testid="circle"
    >
      <div
        class="text text_type_input text_color_input mb-4 absolute head string"
        data-cy="head"
        data-testid="head"
      >
        a
      </div>
      <div
        class="circle   default"
        data-cy="circle"
        data-testid="middleDiv"
      >
        <p
          class="text text_type_circle text_color_input letter"
          data-testid="letter"
        />
      </div>
      <p
        class="text text_type_input text_color_input mt-4 absolute index"
        data-testid="index"
      />
      <div
        class="text text_type_input text_color_input mt-4 absolute tail30 element"
        data-cy="tail"
        data-testid="tail"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`circle tests with index 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="content "
        data-testid="circle"
      >
        <div
          class="text text_type_input text_color_input mb-4 absolute head element"
          data-cy="head"
          data-testid="head"
        />
        <div
          class="circle   default"
          data-cy="circle"
          data-testid="middleDiv"
        >
          <p
            class="text text_type_circle text_color_input letter"
            data-testid="letter"
          />
        </div>
        <p
          class="text text_type_input text_color_input mt-4 absolute index"
          data-testid="index"
        >
          5
        </p>
        <div
          class="text text_type_input text_color_input mt-4 absolute tail60 element"
          data-cy="tail"
          data-testid="tail"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="content "
      data-testid="circle"
    >
      <div
        class="text text_type_input text_color_input mb-4 absolute head element"
        data-cy="head"
        data-testid="head"
      />
      <div
        class="circle   default"
        data-cy="circle"
        data-testid="middleDiv"
      >
        <p
          class="text text_type_circle text_color_input letter"
          data-testid="letter"
        />
      </div>
      <p
        class="text text_type_input text_color_input mt-4 absolute index"
        data-testid="index"
      >
        5
      </p>
      <div
        class="text text_type_input text_color_input mt-4 absolute tail60 element"
        data-cy="tail"
        data-testid="tail"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`circle tests with letter 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="content "
        data-testid="circle"
      >
        <div
          class="text text_type_input text_color_input mb-4 absolute head element"
          data-cy="head"
          data-testid="head"
        />
        <div
          class="circle   default"
          data-cy="circle"
          data-testid="middleDiv"
        >
          <p
            class="text text_type_circle text_color_input letter"
            data-testid="letter"
          >
            a
          </p>
        </div>
        <p
          class="text text_type_input text_color_input mt-4 absolute index"
          data-testid="index"
        />
        <div
          class="text text_type_input text_color_input mt-4 absolute tail30 element"
          data-cy="tail"
          data-testid="tail"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="content "
      data-testid="circle"
    >
      <div
        class="text text_type_input text_color_input mb-4 absolute head element"
        data-cy="head"
        data-testid="head"
      />
      <div
        class="circle   default"
        data-cy="circle"
        data-testid="middleDiv"
      >
        <p
          class="text text_type_circle text_color_input letter"
          data-testid="letter"
        >
          a
        </p>
      </div>
      <p
        class="text text_type_input text_color_input mt-4 absolute index"
        data-testid="index"
      />
      <div
        class="text text_type_input text_color_input mt-4 absolute tail30 element"
        data-cy="tail"
        data-testid="tail"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`circle tests with react element in head 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="content "
        data-testid="circle"
      >
        <div
          class="text text_type_input text_color_input mb-4 absolute head element"
          data-cy="head"
          data-testid="head"
        >
          <div
            class="content "
            data-testid="circle"
          >
            <div
              class="text text_type_input text_color_input mb-4 absolute head element"
              data-cy="head"
              data-testid="head"
            />
            <div
              class="circle   default"
              data-cy="circle"
              data-testid="middleDiv"
            >
              <p
                class="text text_type_circle text_color_input letter"
                data-testid="letter"
              >
                a
              </p>
            </div>
            <p
              class="text text_type_input text_color_input mt-4 absolute index"
              data-testid="index"
            />
            <div
              class="text text_type_input text_color_input mt-4 absolute tail30 element"
              data-cy="tail"
              data-testid="tail"
            />
          </div>
        </div>
        <div
          class="circle   default"
          data-cy="circle"
          data-testid="middleDiv"
        >
          <p
            class="text text_type_circle text_color_input letter"
            data-testid="letter"
          />
        </div>
        <p
          class="text text_type_input text_color_input mt-4 absolute index"
          data-testid="index"
        />
        <div
          class="text text_type_input text_color_input mt-4 absolute tail30 element"
          data-cy="tail"
          data-testid="tail"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="content "
      data-testid="circle"
    >
      <div
        class="text text_type_input text_color_input mb-4 absolute head element"
        data-cy="head"
        data-testid="head"
      >
        <div
          class="content "
          data-testid="circle"
        >
          <div
            class="text text_type_input text_color_input mb-4 absolute head element"
            data-cy="head"
            data-testid="head"
          />
          <div
            class="circle   default"
            data-cy="circle"
            data-testid="middleDiv"
          >
            <p
              class="text text_type_circle text_color_input letter"
              data-testid="letter"
            >
              a
            </p>
          </div>
          <p
            class="text text_type_input text_color_input mt-4 absolute index"
            data-testid="index"
          />
          <div
            class="text text_type_input text_color_input mt-4 absolute tail30 element"
            data-cy="tail"
            data-testid="tail"
          />
        </div>
      </div>
      <div
        class="circle   default"
        data-cy="circle"
        data-testid="middleDiv"
      >
        <p
          class="text text_type_circle text_color_input letter"
          data-testid="letter"
        />
      </div>
      <p
        class="text text_type_input text_color_input mt-4 absolute index"
        data-testid="index"
      />
      <div
        class="text text_type_input text_color_input mt-4 absolute tail30 element"
        data-cy="tail"
        data-testid="tail"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`circle tests with react element in tail 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="content "
        data-testid="circle"
      >
        <div
          class="text text_type_input text_color_input mb-4 absolute head element"
          data-cy="head"
          data-testid="head"
        />
        <div
          class="circle   default"
          data-cy="circle"
          data-testid="middleDiv"
        >
          <p
            class="text text_type_circle text_color_input letter"
            data-testid="letter"
          />
        </div>
        <p
          class="text text_type_input text_color_input mt-4 absolute index"
          data-testid="index"
        />
        <div
          class="text text_type_input text_color_input mt-4 absolute tail30 element"
          data-cy="tail"
          data-testid="tail"
        >
          <div
            class="content "
            data-testid="circle"
          >
            <div
              class="text text_type_input text_color_input mb-4 absolute head element"
              data-cy="head"
              data-testid="head"
            />
            <div
              class="circle   default"
              data-cy="circle"
              data-testid="middleDiv"
            >
              <p
                class="text text_type_circle text_color_input letter"
                data-testid="letter"
              >
                b
              </p>
            </div>
            <p
              class="text text_type_input text_color_input mt-4 absolute index"
              data-testid="index"
            />
            <div
              class="text text_type_input text_color_input mt-4 absolute tail30 element"
              data-cy="tail"
              data-testid="tail"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="content "
      data-testid="circle"
    >
      <div
        class="text text_type_input text_color_input mb-4 absolute head element"
        data-cy="head"
        data-testid="head"
      />
      <div
        class="circle   default"
        data-cy="circle"
        data-testid="middleDiv"
      >
        <p
          class="text text_type_circle text_color_input letter"
          data-testid="letter"
        />
      </div>
      <p
        class="text text_type_input text_color_input mt-4 absolute index"
        data-testid="index"
      />
      <div
        class="text text_type_input text_color_input mt-4 absolute tail30 element"
        data-cy="tail"
        data-testid="tail"
      >
        <div
          class="content "
          data-testid="circle"
        >
          <div
            class="text text_type_input text_color_input mb-4 absolute head element"
            data-cy="head"
            data-testid="head"
          />
          <div
            class="circle   default"
            data-cy="circle"
            data-testid="middleDiv"
          >
            <p
              class="text text_type_circle text_color_input letter"
              data-testid="letter"
            >
              b
            </p>
          </div>
          <p
            class="text text_type_input text_color_input mt-4 absolute index"
            data-testid="index"
          />
          <div
            class="text text_type_input text_color_input mt-4 absolute tail30 element"
            data-cy="tail"
            data-testid="tail"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`circle tests with tail 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="content "
        data-testid="circle"
      >
        <div
          class="text text_type_input text_color_input mb-4 absolute head element"
          data-cy="head"
          data-testid="head"
        />
        <div
          class="circle   default"
          data-cy="circle"
          data-testid="middleDiv"
        >
          <p
            class="text text_type_circle text_color_input letter"
            data-testid="letter"
          />
        </div>
        <p
          class="text text_type_input text_color_input mt-4 absolute index"
          data-testid="index"
        />
        <div
          class="text text_type_input text_color_input mt-4 absolute tail30 string"
          data-cy="tail"
          data-testid="tail"
        >
          b
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="content "
      data-testid="circle"
    >
      <div
        class="text text_type_input text_color_input mb-4 absolute head element"
        data-cy="head"
        data-testid="head"
      />
      <div
        class="circle   default"
        data-cy="circle"
        data-testid="middleDiv"
      >
        <p
          class="text text_type_circle text_color_input letter"
          data-testid="letter"
        />
      </div>
      <p
        class="text text_type_input text_color_input mt-4 absolute index"
        data-testid="index"
      />
      <div
        class="text text_type_input text_color_input mt-4 absolute tail30 string"
        data-cy="tail"
        data-testid="tail"
      >
        b
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
